docs/proposals/sysctl.md

```diff
@@ -0,0 +1,251 @@
+1. to allow special memory management like Java hugepages.
+
+### Use Case: Set kernel parameters to enable kernel features
+
+As an administrator I would like to enable kernel features for certain
+applications while leaving them off for all other containers on a host
+
+1. to enable containerized execution of special purpose applications without
+  the need to enable those kernel features host wide, e.g. ip forwarding for
+  network router daemons
+
+## Community Work
+
+### Runc support for sysctl
+
+Supported sysctls (whitelist) as of RunC 0.1.1 (compare
+[libcontainer config validator](https://github.com/opencontainers/runc/blob/master/libcontainer/configs/validate/validator.go#L107)):
+
+- IPC namespace
+  - System V: `kernel.msgmax`, `kernel.msgmnb`, `kernel.msgmni`, `kernel.sem`,
+    `kernel.shmall`, `kernel.shmmax`, `kernel.shmmni`, `kernel.shm_rmid_forced`
+  - POSIX queues: `fs.mqueue.*`
+- network namespace: `net.*`
+
+Applied changes:
+
+* https://github.com/opencontainers/runc/pull/73
+* https://github.com/opencontainers/runc/pull/303
+
+### Docker support for sysctl
+
+Supported sysctls (whitelist) as of Docker 1.12.0:
+
+- IPC namespace
+  - System V: `kernel.msgmax`, `kernel.msgmnb`, `kernel.msgmni`, `kernel.sem`,
+    `kernel.shmall`, `kernel.shmmax`, `kernel.shmmni`, `kernel.shm_rmid_forced`
+  - POSIX queues: `fs.mqueue.*`
+- network namespace: `net.*`
+
+Applied changes:
+
+* https://github.com/docker/docker#19265
+* https://github.com/docker/engine-api#38
+
+Issues:
+
+* https://github.com/docker/docker#21126
+* https://github.com/ibm-messaging/mq-docker#13
+
+### Rocket support for sysctl
+
+TBD
+
+## Proposed Design
+
+### Sysctl API Resource
+
+We should have an API resource to describe sysctl parameters.
+
+```go
+// Sysctl defines a kernel parameter to be set
+type Sysctl struct {
+	// Name of a property to set
+	Name string `json:"name"`
+	// Value of a property to set
+	Value string `json:"value"`
+}
+```
+
+### Pod API changes
+
+Container specification must be changed to allow the specification of kernel parameters:
+
+```go
+// PodSecurityContext holds pod-level security attributes and common container settings.
+// Some fields are also present in container.securityContext.  Field values of
+// container.securityContext take precedence over field values of PodSecurityContext.
+type PodSecurityContext struct {
+    ...
+	// Sysctls hold a list of namespaced sysctls used for the pod. They are only allowed
+	// if the respective host namespace setting pod.spec.hostIPC or pod.spec.hostNetwork
+	// are false.
+	Sysctls []Sysctl `json:"sysctls,omitempty"`
+}
+```
+
+Note that sysctls must be on the pod level because containers in a pod share IPC and network namespaces (if pod.spec.hostIPC and pod.spec.hostNetwork is false) and therefore cannot have conflicting sysctl values. Moreover, note that all namespaced sysctl supported by Docker/RunC are either in the IPC or network namespace.
+
```

Comment:
Isn't that exactly the existing section? Maybe I don't understand what you mean.